oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/FirewallPolicy.java

```diff
@@ -0,0 +1,190 @@
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class FirewallPolicy {
+
+   public abstract String id();
+
+   public abstract String name();
+
+   @Nullable
+   public abstract String description();
+
+   @Nullable
+   public abstract String state();
+
+   @Nullable
+   public abstract String creationDate();
+
+   @Nullable
+   public abstract String defaultState();
+
+   public abstract List<Rule> rules();
+
+   public abstract List<ServerIp> serverIps();
+
+   @Nullable
+   public abstract String cloudpanelId();
+
+   @SerializedNames({"id", "name", "description", "state", "creation_date", "default", "rules", "server_ips", "cloudpanel_id"})
+   public static FirewallPolicy create(String id, String name, String description, String state, String creationDate, String defaultState, List<Rule> rules, List<ServerIp> serverIps, String cloudpanelId) {
+      return new AutoValue_FirewallPolicy(id, name, description, state, creationDate, defaultState, rules == null ? ImmutableList.<Rule>of() : rules, serverIps == null ? ImmutableList.<ServerIp>of() : serverIps, cloudpanelId);
+   }
+
+   @AutoValue
+   public abstract static class CreateFirewallPolicy {
+
+      public abstract String name();
+
+      @Nullable
+      public abstract String description();
+
+      public abstract List<Rule.CreatePayload> rules();
+
+      @SerializedNames({"name", "description", "rules"})
+      public static CreateFirewallPolicy create(final String name, final String description, List<Rule.CreatePayload> rules) {
+         return new AutoValue_FirewallPolicy_CreateFirewallPolicy(name, description, rules == null ? ImmutableList.<Rule.CreatePayload>of() : rules);
+      }
+   }
+
+   @AutoValue
+   public abstract static class UpdateFirewallPolicy {
+
+      @Nullable
+      public abstract String name();
+
+      @Nullable
+      public abstract String description();
+
+      @SerializedNames({"name", "description"})
+      public static UpdateFirewallPolicy create(final String name, final String description) {
+         return new AutoValue_FirewallPolicy_UpdateFirewallPolicy(name, description);
+      }
+   }
+
+   @AutoValue
+   public abstract static class ServerIp {
+
+      public abstract String id();
+
+      public abstract String ip();
+
+      public abstract String serverName();
+
+      @SerializedNames({"id", "ip", "server_name"})
+      public static ServerIp create(String id, String ip, String serverName) {
+         return new AutoValue_FirewallPolicy_ServerIp(id, ip, serverName);
+      }
+
+      @AutoValue
+      public abstract static class CreateServerIp {
+
+         public abstract List<String> serverIps();
+
+         @SerializedNames({"server_ips"})
+         public static CreateServerIp create(final List<String> serverIps) {
+            return new AutoValue_FirewallPolicy_ServerIp_CreateServerIp(serverIps == null ? ImmutableList.<String>of() : serverIps);
```

Comment:
Same here. Use `copyOf` for lists when present to enforce immutability. Apply this pattern to all domain classes.